src/main/kotlin/org/akuleshov7/utils/Config.kt

```diff
@@ -0,0 +1,89 @@
+package org.akuleshov7.utils
+
+import org.akuleshov7.stat.PROGRAM_NAME
+import org.akuleshov7.stat.StatType
+
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+import kotlinx.cli.required
+
+@Suppress("TOO_LONG_FUNCTION")
+data class Config(
+        val credentials: Credentials?,
+        val repositories: String?,
+        val organizations: String?,
+        val isExtended: Boolean,
+        val isDebug: Boolean,
+        val statType: StatType,
+        val configPath: String?
+) {
+    companion object {
+        fun create(args: Array<String>): Config {
+            // ============ args parsing ===========
+
+            val parser = ArgParser(PROGRAM_NAME)
+
+            val type by parser.option(
+                    ArgType.Choice<StatType>(),
+                    shortName = "t",
+                    description = "Type of statistics"
+            ).required()
+
+            val auth by parser.option(
+                    ArgType.String,
+                    shortName = "a",
+                    description = """GitHub username and authorization token separated by ':'
+                | (can be used to extend the number of requests)""".trimMargin()
```

Comment:
Maybe, provide some info about token requirements (scopes, permissions)?

Reply:
don't get what you mean

Reply:
Github tokens can have [different scopes](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) and give token bearers different permission (read only or write). You should specify required settings for token for your app.

Reply:
we don't need any permissions

Reply:
Ok, but if I want to use git-stat with token and I open [token creation page](https://github.com/settings/tokens) what should I choose? If I generate token with minimum permissions, will it be enough, or will I have to regenerate it again? I'd feel confused in this situation.